src/tree.py

```diff
@@ -0,0 +1,91 @@
+import math
+
+from carcassonne_engine.models import SerializedGameWithID
+from carcassonne_engine.requests import MoveWithState
+
+from .agents import Agent
+from .dispatch import GameDispatch
+from .tokens import CancelationToken
+from .utils import upper_confidence_bound
+
+DISCOUNT_FACTOR = 0.9
+EXPLORATION_WEIGHT = 0.01
+
+
+class Tree:
+    def __init__(
+        self, agent: Agent, initial_game: SerializedGameWithID, dispatch: GameDispatch
+    ) -> None:
+        self.initial_game = initial_game
+        states = dispatch.gen_initial_states(
+            initial_game.id, initial_game.game.current_tile
+        )
+        prob = [1.0 for _ in states]
```

Comment:
Is ignoring the actual probabilities preferred here? They do differ depending on how many times each tile template occurs in the tileset.

Reply:
In this case moves are possible placements of current tiles so each one is equally probable

Reply:
Makes sense, I overlooked that we already know the current tile and these are probabilities of the placements, not the probabilities of the tile that you'll get from the remaining stack.